File: 백준/Silver/1475. 방 번호/방 번호.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String n = br.readLine();

        int[] count = new int[10];

        for (int i = 0; i < n.length(); i++) {
            int digit = n.charAt(i) - '0';
            count[digit]++;
        }

        int sixNineCount = count[6] + count[9];
        count[6] = count[9] = (sixNineCount + 1) / 2;

        int max = 0;
        for (int i = 0; i < 10; i++) {
            max = Math.max(max, count[i]);
        }

        System.out.print(max);
    }
}

File: 백준/Silver/1475. 방 번호/방 번호.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n = readLine()
    val arr = IntArray(10)
    for (i in n) {
        arr[i - '0']++
    }
    var tmp = 0
    for (i in 0..9) {
        if (i != 6 && i != 9) {
            if (arr[i] > tmp) {
                tmp = arr[i]
            }
        }
    }
    val cnt = (arr[6] + arr[9] + 1) / 2
    if (tmp >= cnt) {
        println(tmp)
    } else {
        println(cnt)
    }
}